nucleus/cluster/ssh/src/main/java/org/glassfish/cluster/ssh/launcher/JavaSystemJschLogger.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright (c) 2023 Contributors to the Eclipse Foundation
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0, which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the
+ * Eclipse Public License v. 2.0 are satisfied: GNU General Public License,
+ * version 2 with the GNU Classpath Exception, which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ */
+package org.glassfish.cluster.ssh.launcher;
+
+import com.jcraft.jsch.JSch;
+import java.lang.System.Logger;
+import java.lang.System.Logger.Level;
+
+public class JavaSystemJschLogger implements com.jcraft.jsch.Logger {
+
+    private final Logger logger;
+
+    public JavaSystemJschLogger() {
+        this.logger = System.getLogger(JSch.class.getName());
+    }
+
+    public JavaSystemJschLogger(String loggerName) {
+        this.logger = System.getLogger(loggerName);
+    }
+
+    @Override
+    public boolean isEnabled(int jschLevel) {
+        return logger.isLoggable(getSystemLoggerLevel(jschLevel));
+    }
+
+    @Override
+    public void log(int jschLevel, String message) {
+        logger.log(getSystemLoggerLevel(jschLevel), message);
+    }
+
+    @Override
+    public void log(int jschLevel, String message, Throwable cause) {
+        if (cause != null) {
+            logger.log(getSystemLoggerLevel(jschLevel), message, cause);
+        } else {
+            logger.log(getSystemLoggerLevel(jschLevel), message);
+        }
+    }
+
+    static Level getSystemLoggerLevel(int jschLevel) {
+        switch (jschLevel) {
+            case DEBUG:
+                return Level.TRACE;
+            case INFO:
+                return Level.DEBUG;
+            case WARN:
```

Comment:
WARN to INFO and INFO to DEBUG sounds suspicious ... I don't know how those logs look like, but I would expect that JSCH is based on some old presumptions about FATAL in LOG4J or Commons Logging or where I have seen it for the last time some 15 years ago ...

Reply:
Yes, it's suspicious, but reasonable. Have a look at this log file: [gf-jsch-ping-node.log](https://github.com/eclipse-ee4j/glassfish/files/11750634/gf-jsch-ping-node.log), which contains messages from jsch just for a single ping action to an SSH node:

* 1 INFO message
* 47 FINE messages
* 6 FINER messages

If we mapped WARN to WARNING, INFO to INFO, DEBUG to DEBUG, it would be 1 WARNING, 47 INFO messages and 6 FINE messages. Too many INFO messages for a single SSH connection, IMHO. And the warning message just says that the host was added to the list of known hosts, which, according to me, doesn't qualify as a WARNING but rather as an INFO message. It's logged for every connection, because the list of known hosts isn't preserved between sessions.

Does it now make sense why I made this shift of log levels? :smiley:

Reply:
To clarify this in the code, I added a comment in this logger class to clarify how the JSCH log levels are mapped to levels in GlassFish and why.

Reply:
Seems reasonable, imo

Reply:
Ok, that's why I asked, thanks.